tests/test_target.py

```diff
@@ -1,11 +1,121 @@
+from dissect.target.filesystem import VirtualFilesystem
 from dissect.target.filesystems.dir import DirectoryFilesystem
-from dissect.target.target import Event, Target
+from dissect.target.loader import LOADERS
+from dissect.target.loaders.raw import RawLoader
+from dissect.target.target import Event, Target, TargetLogAdapter, log
+
+
+class ErrorCounter(TargetLogAdapter):
+    errors = 0
+
+    def error(*args, **kwargs):
+        ErrorCounter.errors += 1
+
+
+@pytest.mark.parametrize(
+    "topo, entry_point, expected_result, expected_errors",
+    [
+        # Base cases:
+        # Scenario 0:  Can we still load a single target that's not a dir?
+        (["/raw.tst"], "/raw.tst", "[TestLoader('/raw.tst')]", 0),
+        # Scenario 0b: Make sure we still get so see errors if a target fails to load
+        # (1 error + 1 exception no load)
+        (["/raw.vbox"], "/raw.vbox", "[]", 2),
+        # Scenario 0c: Attempting to load a dir with nothing of interest yield an error
+        # (1 exception no load)
+        (["/dir/garbage"], "/dir", "[]", 1),
+        # Dir cases:
+        # Scenario 1:  Dir is loadable target (simple)
+        (["/dir/etc", "/dir/var"], "/dir", "[DirLoader('/dir')]", 0),
+        # Scenario 2: dir contains multiple loadable targets
+        (["/dir/raw.img", "/dir/raw2.tst"], "/dir", "[RawLoader('/dir/raw.img'), TestLoader('/dir/raw2.tst')]", 0),
+        # Scenario 2b: dir contains multiple loadable targets and some garbage
+        # (it will wrap the garbage in a RawLoader to try)
+        (
+            ["/dir/raw.img", "/dir/raw2.tst", "/dir/info.txt"],
+            "/dir",
+            "[RawLoader('/dir/raw.img'), TestLoader('/dir/raw2.tst')," " RawLoader('/dir/info.txt')]",
+            0,
+        ),
+        # Scenario 3: dir contains 1 loadable dir as well as 2 loadable targets
+        (
+            ["/dir/unix/etc", "/dir/unix/var", "/dir/raw.img", "/dir/raw2.tst"],
+            "/dir",
+            "[DirLoader('/dir/unix'), " "RawLoader('/dir/raw.img'), " "TestLoader('/dir/raw2.tst')]",
+            0,
+        ),
+        # Scenario 3b: dir contains 2 loadable dirs as well as 2 loadable targets
+        (
+            [
+                "/dir/unix1/etc",
+                "/dir/unix1/var",
+                "/dir/unix2/etc",
+                "/dir/unix2/var",
+                "/dir/raw.img",
+                "/dir/raw2.tst",
+            ],
+            "/dir",
+            "[DirLoader('/dir/unix1'), DirLoader('/dir/unix2'),"
+            " RawLoader('/dir/raw.img'), TestLoader('/dir/raw2.tst')]",
+            0,
+        ),
+        # Scenario 4: dir is a loadable dir but contains other files including loadable targets
+        (
+            [
+                "/dir/etc",
+                "/dir/var",
+                "/dir/raw.img",
+                "/dir/info.txt",
+            ],
+            "/dir",
+            "[DirLoader('/dir')]",
+            0,
+        ),
```

Comment:
Can you add a scenario for Windows directory targets? So `/dir/c:/windows/system32`. Because here both `/dir` and `/dir/c:` are valid `DirLoader` targets because it supports Windows-like directories with and without directory letters.
```

Reply:
Added win-scenarios